pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -595,6 +597,30 @@ def sync(self, options=None):
             )
         )
 
+    def lock(self, duration=LOCK_EXPIRATION_OFFSET, retry_wait=AWAIT_ACTIVATION_SLEEP):
+        """
+        Lock the repo to show other tasks that it is being worked on.
+        This does not prevent modifications to the repo with the Pulp API, it
+        just provides a way to pause tasks to prevent race conditions.
+
+        Args:
+            duration
+                How long the lock should remain valid for in seconds.
+            retry_wait:
+                How long to wait between attempting to activate the lock in
+                seconds.
+        """
+        return RepoLock(
```

Comment:
Sorry I haven't realized this earlier, but I think we ought to have a kind of 'name' or 'context' string field as well which is stored on the lock.  It's just for logging and debugging purposes.

The string would be provided by the caller and should briefly describe why you have the lock, for example:

```python
with repo.lock('updating maintenance report'):
    # ...the steps to fetch, modify and re-upload maintenance report
```

It doesn't have to be unique across all existing locks, but ideally is unique for each code path obtaining a lock.

This wouldn't affect the locking mechanism, the point of it is just to make it easier to trace who obtained a lock. 

e.g. imagine we're facing a problem that some tasks are running slowly because there often seems to be some stale locks on a repo.  We need to figure out who is creating those locks and failing to release them.

If the locks included a field like `{"context": "updating maintenance report"}`, we could simply grep for `updating maintenance report` across our code and immediately find the code we should start debugging.  Otherwise we would need to check every usage of .lock(), cross-reference lock IDs with logs in Splunk (hoping all relevant log messages made it there), and so on.

Also, the log messages for acquiring/releasing/deleting stale locks could include this string in the message.  That'll be useful in some cases, e.g. when we delete a stale lock, if we generated a message like:

```
WARNING: removing expired lock pulplib-lock-claim-abc123 (updating maintenance report)
```

Then anyone reading the log could immediately know that removing this expired lock might affect other tasks relating to "updating maintenance report" and shouldn't affect anything else.